server/controllers/work/editWork.js

```diff
@@ -0,0 +1,39 @@
+const { Work } = require('../../database/models');
+const { customError } = require('../errors');
+const { workEditValidation } = require('../../utils/validation');
+const { uploadImage, deleteImage } = require('../../utils');
+
+const editWork = async (req, res, next) => {
+  try {
+    const { id } = await workEditValidation.validateAsync(req.body);
+    const { title, content } = req.body;
+    const { providerID } = req;
+    const work = await Work.findByPk(id);
+    if (!work) {
+      throw customError('work does not exist', 400);
+    }
+    if (work.userId !== +providerID) {
+      throw customError('Unauthorized', 401);
+    }
+    let imgUrl = '';
+    if (req.files) {
+      const { image } = req.files;
+      imgUrl = await uploadImage(image, work.userId);
+      deleteImage(work.image);
```

Comment:
move this after sending the response, at the end of the controller so we don't delay user from getting a response